### ISSUE TITLE
FN — `ExprId` allocation-site keys are per-function, so unrelated bodies collide in one heap

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#134]
Found while sizing #94 (its blocker is the flat `SummaryValue`, and this sits underneath it).

## Shape

`Heap` is `HashMap<ExprId, HeapValue>` and `ExprId` is documented as the *allocation-site key*. But `BlockBuilder::expr_counter` starts at `0` in **every** `BlockBuilder`, and a fresh builder is made for every function body (`build_fn_body_cfg`, `build_expr_fn_body_cfg`, `build_stmts_cfg`). `remap_cfg` remaps `HookLabel`s on splice and leaves `ExprId`s untouched.

So a component's render body, each of its effect/callback bodies, and every inlined callee all number their allocation sites from zero into **one shared heap**. The later `Heap::insert` wins.

Probe on the lowered IR:

```
render_cfg alloc ids: [ExprId(0), ExprId(1), ExprId(4)]
effect 0   alloc ids: [ExprId(0)]                        ← collides with `bagA`
```

## It is a false negative, not just noise

Both directions reproduce end-to-end. The two fixtures differ **only** by an unrelated effect elsewhere in the same component.

**FP direction** — a stable prefix is lost:

```jsx
// silent (correct)                          | fires `missing-deps` (wrong)
function C() {                               | function C({ p }) {
  const r = useRef(0);                       |   const r = useRef(0);
  const bag = { r };                         |   const bag = { r };
  const cb = useCallback(() => bag.r.current, []);
  return <div onClick={cb} />;               |   useEffect(() => {
}                                            |     const shadow = { r: { current: p } };
                                             |     console.log(shadow);
                                             |   }, [p]);
                                             |   return <div onClick={cb} />;
                                             | }
```

**FN direction — the forbidden one** — a genuine stale closure disappears:

```jsx
// fires `missing-deps` (correct)            | SILENT (wrong)
function C({ p }) {                          | function C({ p }) {
  const bag = { inner: { v: p } };           |   const r = useRef(0);
  const cb = useCallback(() => bag.inner.v, []);
  return <div onClick={cb} />;               |   const bag = { inner: { v: p } };
}                                            |   const cb = useCallback(() => bag.inner.v, []);
                                             |   useEffect(() => {
                                             |     const shadow = { inner: r };   // ← overwrites bag's heap entry
                                             |     console.log(shadow);
                                             |   }, [p]);
                                             |   return <div onClick={cb} />;
                                             | }
```

`bag.inner` is a per-render object, so the `[]` closure holds a stale one and must fire. The effect's `{ inner: r }` lands at the same `ExprId`, overwrites the entry, and `bag.inner` reads as the `useRef` — stable, silent.

Anything that reads members through the heap is affected: `missing-deps` (ADR-040 prefixes), the binding chase in `engine/seeds.rs` (ADR-033 §2 selects through an `ObjectLit`), `resolve_locs`, prop maps.

## Fix direction

Give `ExprId` the uniqueness its own doc claims, the way `HookLabel` already has it:

1. **Per-file counter** — one counter shared by every `BlockBuilder` of a `lower_program` run, threaded exactly as `smap` already is (~12 call sites). Fixes render-vs-nested-body.
2. **Remap on splice** — `remap_cfg` offsets `ExprId`s alongside `HookLabel`s, so an inlined callee (and the *same* callee inlined at two call sites) gets its own range. Fixes cross-file and repeated inlining.

Expect the corpus to move in both directions: FPs removed where a prefix was wrongly clobbered, and findings **added** where one was wrongly silenced.
